### PR TITLE
[match] fix devices fetch for tvOS platform

### LINKED ISSUE #21825: [tvOS] Match: A parameter 'filter[platform]' has an invalid value : 'Empty filter values are not supported.
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
Match fails for tvOS platform.

##### Command executed

```
match(
    type: "appstore",
    platform: "tvos",
)
```

##### Complete output when running fastlane, including the stack trace and command used

```
+------------------------------+
|         Lane Context         |
+------------------+-----------+
| DEFAULT_PLATFORM | ios       |
| PLATFORM_NAME    | tvos      |
| LANE_NAME        | tvos beta |
+------------------+-----------+
[23:14:36]: {
  "errors" : [ {
    "id" : "f4d67a04-e1db-477b-a9db-9af7d3beb0d6",
    "status" : "400",
    "code" : "PARAMETER_ERROR.INVALID",
    "title" : "A parameter has an invalid value",
    "detail" : "A parameter 'filter[platform]' has an invalid value : 'Empty filter values are not supported.'",
    "source" : {
      "parameter" : "filter[platform]"
    }
  } ]
}
+-------------------------------------------------+
|                fastlane summary                 |
+------+----------------------------+-------------+
| Step | Action                     | Time (in s) |
+------+----------------------------+-------------+
| 1    | Verifying fastlane version | 0           |
| 2    | default_platform           | 0           |
| 3    | app_store_connect_api_key  | 0           |
| 💥   | match                      | 3           |
+------+----------------------------+-------------+
[23:14:36]: fastlane finished with errors### Environment
```

fastlane 2.219.0

### PR DESCRIPTION
**Pull request by nekrich**
### Checklist

- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.
- [x] I've added or updated relevant unit tests.

### Motivation and Context

Resolves #21825.

### Description

1. Add check for :tvos platform.
2. Skip adding filter if device_platforms is empty

### CODE REVIEW

File: spaceship/lib/spaceship/connect_api/models/device.rb [hunk 1]
```diff
@@ -106,9 +108,9 @@ def self.devices_for_platform(platform: nil, include_mac_in_profiles: false, cli
         end
 
         filter = {
-          status: Spaceship::ConnectAPI::Device::Status::ENABLED,
-          platform: device_platforms.uniq.join(',')
+          status: Spaceship::ConnectAPI::Device::Status::ENABLED
         }
+        filter[:platform] = device_platforms.join(',') unless device_platforms.empty?
```
Comment (lacostej) [inline spaceship/lib/spaceship/connect_api/models/device.rb:113]: we don't need the `uniq` anymore?

Comment (nekrich) [reply spaceship/lib/spaceship/connect_api/models/device.rb:113]: Forgot about this. Will fix in a couple of minutes

**Review by lacostej — APPROVED** [review]
LGTM :)

**Comment by tache** [comment]
This PR worked for me. TNX

**Comment by buscarini** [comment]
When is this PR going to be merged? I'm having this issue. Is there an easy way to install fastlane with this change applied? thanks!

**Comment by redreceipt** [comment]
What's the process/status on this PR being merged and included in a release? we need the fix as well

**Comment by lacostej** [comment]
> When is this PR going to be merged? I'm having this issue. Is there an easy way to install fastlane with this change applied? thanks!

This was merged.  if you use a Gemfile, you can easily point to the `master`  branch of fastlane until we make a release, which hopefully should happen soon. We have one side issue to clarify.

**Comment by MartinGonzalez** [comment]
Hi, I still have this issue on version `2.219.0`. I'm using sigh

```
  "errors" : [ {
    "id" : "a823ac2d-48e3-4063-8248-a45b1156f503",
    "status" : "400",
    "code" : "PARAMETER_ERROR.INVALID",
    "title" : "A parameter has an invalid value",
    "detail" : "A parameter 'filter[platform]' has an invalid value : 'Empty filter values are not supported.'",
    "source" : {
      "parameter" : "filter[platform]"
    }
  } ]
}
```

**Pull request merged**